### ISSUE TITLE
talos.inline.config not working in Omni generated VMWare ova image

### ISSUE THREAD

**Issue by sedh-sbab** [siderolabs/talos#9503]
## Bug Report

When I generate a Talos `v1.8.1` image for VMWare platform using `omnictl` our CA Certificate config is ignored when using the `talos.config.inline` kernel args and we get x509 Certificate errors in the console log of the Talos VM when trying to connect to our Omni instance.

I provide `--extra-kernel-args talos.config.inline=${TALOS_CONFIG_INLINE}` to `omnictl` where `$TALOS_CONFIG_INLINE` is created by using the guide here https://www.talos.dev/v1.8/reference/kernel/#talosconfiginline. The config document is a CA certificate, see https://www.talos.dev/v1.8/talos-guides/configuration/certificate-authorities/#appending-the-certificate-authority. I have tried using the offical `factory.talos.dev` with the same result. I have checked the GRUB menu and the `talos.config.inline` key and value is present.

If I instead provide the same CA certificate config document as a base64 encoded string and instead use the VMware guestinfo the CA certificate works great and the node can connect to our Omni instance without any errors. I use this command to insert the config document to the VM host,
```
govc vm.change \
  -e "guestinfo.talos.config=$(cat ca-root-config.yml | base64)"
....
```

I have tried to wipe and reset the machine and edit the kernel arguments to change the platform and remove the `talos.config=guestinfo` line without any luck. But am not sure it has anything to do with this.

Platform: VMWare (OVA template)
Talos Version: v1.8.1

**Comment by smira** [comment]
Please provider kernel logs.

P.S. It's way better to use userdata than `talos.config.inline` with Omni.

**Comment by sedh-sbab** [comment]
The kernel log: the best I can do is an image, hope that works
![image](https://github.com/user-attachments/assets/38b2602f-78de-4bf8-bbdd-32a1e6314877)
The rest of the logs are mostly from time.syncController that can't connect out to internet.

The status of the node stays like this forever:
![image](https://github.com/user-attachments/assets/46dfc4eb-df31-40af-8a8f-ec04dcd14ffb)

P.S For the userdata part, that actually sounds very reasonable, since it don't have quite the same limitations. Thank you.

**Comment by smira** [comment]
We need full kernel logs, (serial console logs) to understand why the config failed to load. We can't debug much without it, sorry.

**Comment by sedh-sbab** [comment]
> We need full kernel logs, (serial console logs) to understand why the config failed to load. We can't debug much without it, sorry.

I'll check if I can attach a serial and save to disk

**Comment by sedh-sbab** [comment]
Here it is! I have redacted the sensitive information.
[console-log.txt](https://github.com/user-attachments/files/17381926/console-log.txt)

On line 15 and 104 the `talos.config.inline` is clearly missing. I can see it in the GRUB menu though.

**Comment by smira** [comment]
If you're booting from the OVA, it should be there, unless there was something else happening (like an upgrade) which would wipe that kernel argument?

**Comment by smira** [comment]
I wonder if it's too big and gets cut by GRUB... maybe your certificate is RSA? ECDSA is way smaller

**Comment by sedh-sbab** [comment]
Well in totalt with our `talos.config.inline` the whole command is 2700 bytes.
```
BOOT_IMAGE=/A/vmlinuz talos.platform=vmware talos.config=guestinfo console=tty0 console=ttyS0 earlyprintk=ttyS0,115200 net.ifnames=0 init_on_alloc=1 slab_nomerge pti=on consoleblank=0 nvme_core.io_timeout=4294967295 printk.devkmsg=on ima_template=ima-ng ima_appraise=fix ima_hash=sha512 siderolink.api=https://<REDACTED>:443?grpc_tunnel=false&jointoken=<REDACTED> talos.events.sink=[fdae:41e4:649b:9303::1]:8091 talos.logging.kernel=tcp://[fdae:41e4:649b:9303::1]:8092 talos.config.line=<2213 bytes>
```

In your documentation it says the Linux kernel args has a max size of 4096, but maybe grub has another limit?

**Comment by smira** [comment]
yes, it might be GRUB or the boot protocol used with GRUB limit (I guess you're booting in BIOS mode on VMWare?)

**Comment by sedh-sbab** [comment]
> yes, it might be GRUB or the boot protocol used with GRUB limit (I guess you're booting in BIOS mode on VMWare?)

Yes, BIOS.
